### PR TITLE
Implement HLE macros for render target clears

### PR DESCRIPTION
**Pull request by gdkchan**
This adds a HLE macro for render target clears (color and depth-stencil). The main advantage here is clearing textures with more than one layer (array or 3D). Before it would clear one layer at a time, with this change it can clear all of them at once, which might result in a minor performance improvement on games that render to array or 3D textures (most games won't).

Testing is welcome to ensure there's no regressions.

### CODE REVIEW

File: Ryujinx.Graphics.Gpu/Engine/MME/MacroHLE.cs [hunk 1]
```diff
@@ -53,6 +59,31 @@ public void Execute(ReadOnlySpan<int> code, IDeviceState state, int arg0)
...
+        }
+
+        /// <summary>
+        /// Clears the current depth-stencil target.
+        /// </summary>
+        /// <param name="state">GPU state at the time of the call</param>
+        /// <param name="arg0">First argument of the call</param>
+        private void ClearDepthStencil(IDeviceState state, int arg0)
+        {
+            int layerCount = state.Read(0x1230);
```
Comment (jduncanator) [inline Ryujinx.Graphics.Gpu/Engine/MME/MacroHLE.cs:82]: Is it possible to define constants for these offsets?

Comment (gdkchan) [reply Ryujinx.Graphics.Gpu/Engine/MME/MacroHLE.cs]: Added.

**Review by Thog — APPROVED** [review]

**Review by riperiperi — APPROVED** [review]
LGTM.

**Comment by LukeWarnut** [comment]
Causes a regression in Game Builder Garage's graphics.

Master:
image

PR:
image

**Comment by gdkchan** [comment]
This is most likely the regression that was fixed by #3525 rather than a regression caused by this PR. The build here is outdated, it was made before the fix got in. I tested Game Builder Garage here and it looks fine after rebase. In any case, I rebased the branch so it should make a new build, and you can retest it if you want.

**Pull request merged**